gems/smithy-client/lib/smithy-client/shapes.rb

```diff
@@ -91,7 +91,30 @@ def member(name)
       class IntegerShape < Shape; end
 
       # Represents an IntEnum shape.
-      class IntEnumShape < EnumShape; end
+      class IntEnumShape < Shape
```

Comment:
Inheriting from EnumShape and StructureShape makes is_a? checks order dependent for visiting. While this is duplicating, it's safer.

Reply:
If we wanted to reduce the duplicaiton, we could do use an `include` with some sort of ShapeWithMembers module or something.  But the duplication is minimal and fine.